### ISSUE TITLE
Postgres returning 0 rows after Mapnik appends LIMIT 0

### ISSUE THREAD

**Issue by PeteDevoy** [mapnik/mapnik-support#70]
Hi folks

I am encountering a number of problems trying to get Mapnik render geometries from a 'building' table I have in PostGIS.  I have tried to reduce everything to the simplest configuration and I haved turned on query logging in Postgres.  I can see that this query is being executed:

`SELECT * FROM building LIMIT 0`

which returns 0 rows.  I thought I may be able to circumvent this by explicitly stating a row_limit in my XML but it did not work, here is my config:

```
1 <Map background-color="#3399ddff" srs="+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717   +x_0=400000 +y_0=-100000 +ellps=airy +towgs84=446.448,-125.157,542.06,0.15,0.247,0.842,-20.489 +units=m +no_defs">
2   <Style name="building">
3     <Rule>
4       <PolygonSymbolizer fill="#f2eff9" />
5       <LineSymbolizer stroke="rgb(50%,50%,50%)" />
6     </Rule>
7   </Style> 
8   <Layer name="building" srs="+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +towgs84=446.448,-125.157,542.06,0.15,0.247,0.842,-20.489 +units=m +no_defs">
9     <StyleName>building</StyleName>
10     <Datasource>
11       <Parameter name="srid">27700</Parameter>
12       <Parameter name="type">postgis</Parameter>
13       <Parameter name="user">peter_devoy</Parameter>
14       <Parameter name="dbname">os</Parameter>
15       <Parameter name="table">building</Parameter>
16       <Parameter name="geometry_field">geom</Parameter>
17       <Parameter name="row_limit">1000</Parameter>
18     </Datasource>
19   </Layer>
20 </Map>
```

`
And, I don't suppose it should matter but here is the python I am using to create the output which I borrowed from the OSM and slightly modified:

```
1 #!/usr/bin/python
2 from mapnik import *
3 import sys, os
4 def drawMap(filename,west,south,east,north):
5     print(filename)
6     sz = 250
7     m = Map(sz,sz,"+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.999601 +x_0=400000 +y_0=-100000 +ellps=airy  +units=m +towgs84=446.448,-125.157,542.060,0.1502,0.2470,0.8421,-20.4894 +units=m +nodefs")
8     bbox = Envelope(west,south,east,north)
9     print bbox
10     load_map(m,"basemap.xml")
11     m.zoom_to_box(bbox)
12     im = Image(sz,sz)
13     render(m, im)
14     render(m, im)
15     view = im.view(0,0,sz,sz) # x,y,width,height
16     view.save(filename,'png')
17 drawMap('preston.png',177705, 34767, 177870, 34957)
```

I am scratching a hole in my head so any ideas would be appreciated!

**Comment by lightmare** [comment]
That's a warm-up query to figure out what columns there are.

You have a typo in Layer srs="... +no    _defs". If it fails to reproject bounding box correctly, rendering queries are pretty likely to return nothing.

**Comment by PeteDevoy** [comment]
Oh, that makes sense, thanks.

Unfortunately that was just a formatting error, I must have accidentally deleted the character when trying to get it into a markdown codeblock.

Is there any other information I can provide to help figure out why I am not getting any results?

This are the only two queries which get logged when I run the script:

```
2016-04-03 16:32:56 BST [1886-1] peter_devoy@os LOG:  statement: SELECT * FROM building LIMIT 0
2016-04-03 16:32:56 BST [1886-2] peter_devoy@os LOG:  statement: SELECT ST_XMin(ext),ST_YMin(ext),ST_XMax(ext),ST_YMax(ext) FROM (SELECT ST_Extent(geom) as ext from building) as tmp.
```

Edit: I just noticed the SRID is 0 in the database, so shp2pgsql didn't pull it in, I am going to update it and see what happens.
Edit 2: It made no difference.  I ran the second query above manually to confirm the output and it does retrieve the correct extent for my data:

```
  st_xmin  | st_ymin  |  st_xmax  |  st_ymax
-----------+----------+-----------+-----------
 199941.99 | 33987.88 | 300056.65 | 100059.07
```

Everything is in the same projection.

An odd thing is that when trying to get TileStache work with the layer it does 'feel' like it's working [labouring] because each tile loads in about one to two seconds and that would match for there are about 220K geometries in the `building` table.

**Comment by PeteDevoy** [comment]
Okay, so after I fixed the SSID the reason I could not see anything was because the 1000 limit was kicking in.  I discovered this after changing the python script to cover the full extent of my data and making the image 10000x10000px.  After removing the 1000 limit it now loads all buildings onto the layer.

Now Mapnik is confirmed working it is time to loose some more hair over TileStache >.<

Thanks @lightmare
